File: Python/underautomation/universal_robots/primary_interface/popup_message_event_args.py
import typing
from underautomation.universal_robots.primary_interface.requested_types import RequestedTypes
from underautomation.universal_robots.common.package_event_args import PackageEventArgs
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.PrimaryInterface import PopupMessageEventArgs as popup_message_event_args

class PopupMessageEventArgs(PackageEventArgs):
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = popup_message_event_args()
		else:
			self._instance = _internal
	@property
	def request_id(self) -> int:
		return self._instance.RequestId
	@request_id.setter
	def request_id(self, value: int):
		self._instance.RequestId = value
	@property
	def requested_type(self) -> RequestedTypes:
		return RequestedTypes(self._instance.RequestedType)
	@requested_type.setter
	def requested_type(self, value: RequestedTypes):
		self._instance.RequestedType = value
	@property
	def warning(self) -> bool:
		return self._instance.Warning
	@warning.setter
	def warning(self, value: bool):
		self._instance.Warning = value
	@property
	def error(self) -> bool:
		return self._instance.Error
	@error.setter
	def error(self, value: bool):
		self._instance.Error = value
	@property
	def blocking(self) -> bool:
		return self._instance.Blocking
	@blocking.setter
	def blocking(self, value: bool):
		self._instance.Blocking = value
	@property
	def popup_message_title(self) -> str:
		return self._instance.PopupMessageTitle
	@popup_message_title.setter
	def popup_message_title(self, value: str):
		self._instance.PopupMessageTitle = value
	@property
	def popup_text_message(self) -> str:
		return self._instance.PopupTextMessage
	@popup_text_message.setter
	def popup_text_message(self, value: str):
		self._instance.PopupTextMessage = value

File: UnderAutomation.UniversalRobots.ObfuscatedSources/PrimaryInterface/PopupMessageEventArgs.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using Common;

namespace PrimaryInterface {
	/// <summary>
	/// Popup message that appears with the Assignment instruction or the URScript popup() function
	/// </summary>
	public class PopupMessageEventArgs : PackageEventArgs {

		/// <summary>
		/// Each popup has a unique ID
		/// </summary>
		public uint RequestId { get; set; }

		/// <summary>
		/// Type for assignment popups
		/// </summary>
		public RequestedTypes RequestedType { get; set; }

		/// <summary>
		/// Popup is a warning
		/// </summary>
		public bool Warning { get; set; }

		/// <summary>
		/// Popup is an error
		/// </summary>
		public bool Error { get; set; }

		/// <summary>
		/// Popup is blocking script execution
		/// </summary>
		public bool Blocking { get; set; }

		/// <summary>
		/// Popup title
		/// </summary>
		public string PopupMessageTitle { get; set; }

		/// <summary>
		/// Popup message, null for assignment popups
		/// </summary>
		public string PopupTextMessage { get; set; }
	}
}
